AzureCore/Source/Pipeline/Policies/LoggingPolicy.swift

```diff
@@ -0,0 +1,133 @@
+//
+//  LoggingPolicy.swift
+//  AzureCore
+//
+//  Created by Brandon Siegel on 10/2/19.
+//  Copyright © 2019 Azure SDK Team. All rights reserved.
+//
+
+import Foundation
+
+public class DefaultLoggingPolicy: PipelineStageProtocol {
+
+    public var next: PipelineStageProtocol?
+    private let log: PipelineLogger
+    private lazy var attachmentRegex = NSRegularExpression("attachment; ?filename=([\"\\w.]+)",
+                                                           options: .caseInsensitive)
+
+    public init(logger: PipelineLogger? = nil) {
+        if let logger = logger {
+            self.log = logger
+        } else if #available(macOS 10.12, iOS 10.0, watchOS 3.0, tvOS 10.0, *) {
+            self.log = PipelineOSLogAdapter()
+        } else {
+            self.log = PipelineNSLogger()
+        }
+    }
+
+    public func onRequest(_ request: inout PipelineRequest) {
+        let req = request.httpRequest
+        log.debug("Request URL: \(req.url)")
+        log.debug("Request method: \(req.httpMethod.rawValue)")
+        log.debug("Request headers:")
+        for (header, value) in req.headers {
+            log.debug("    \(header): \(value)")
+        }
+        log.debug("Request body:")
+        log.debug(loggableBodyFor(headers: req.headers, bodyText: req.text))
```

Comment:
Gives some guidance on when and where prepositions should appear in argument labels:
https://swift.org/documentation/api-design-guidelines/#argument-labels

This seems like the appropriate use, most like the "move(toX:y)" example.

Reply:
Fixed in f976ebb